lib/errors/error-collector.js

```diff
@@ -142,13 +142,18 @@ class ErrorCollector {
     }
 
     const isErroredTransaction = urltils.isError(this.config, transaction.statusCode)
+    const isIgnoredErrorStatusCode = urltils.isIgnoredError(
```

Comment:
There's `urltils.isIgnoredError()` and `helpers.shouldIgnoreError()`. Should these be consolidated? Not saying that should be part of this PR, just came across that when reviewing this section of code.

Reply:
yeah, the error helpers got introduced along the way but the code wasn't really consolidated in meaningful ways.

this all could use a really thoughtful redesign but certainly some of that could be simplified.